### PR TITLE
[SYCL] Reorganize annotated_arg/ptr folder

### PR DESCRIPTION
**Pull request by broxigarchen**
Reorganize annotated_ptr/arg folder

sycl/ext
----oneapi/
--------experimental
--------------annotated_ptr
--------------------annotated_ptr.hpp
--------------annotated_arg
--------------------annotated_arg.hpp
--------------common_annotated_properties
--------------------properties.hpp
----intel/
--------experimental
--------------fpga_annotated_properties.hpp
--------------fpga_kernel_properties.hpp

### CODE REVIEW

File: sycl/include/sycl/ext/intel/experimental/fpga_annotated_properties.hpp [hunk 1]
```diff
@@ -14,7 +14,7 @@
 namespace sycl {
 __SYCL_INLINE_VER_NAMESPACE(_V1) {
 namespace ext {
-namespace oneapi {
+namespace intel {
```
[the remark this answers is not in the document]
Comment (broxigarchen) [reply sycl/include/sycl/ext/intel/experimental/fpga_annotated_properties.hpp:17]: The spec should be intel

https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/proposed/sycl_ext_intel_fpga_kernel_arg_properties.asciidoc

File: sycl/include/sycl/ext/oneapi/common_annotated_properties/properties.hpp
Comment (tiwaria1) [inline sycl/include/sycl/ext/oneapi/common_annotated_properties/properties.hpp:1]: Why is this under ext/oneapi instead of ext/intel?

Comment (broxigarchen) [reply sycl/include/sycl/ext/oneapi/common_annotated_properties/properties.hpp:1]: This is a mistake. Corrected.

For the common properties under oneAPI. For now we have "restrict", but it's not implemented yet.
https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/experimental/sycl_ext_oneapi_kernel_arg_properties.asciidoc

 This is a place holder and also holding some utilities

**Comment by broxigarchen** [comment]
Hi @dm-vodopyanov can you help to review this PR? We want to get this in before CC.

File: sycl/include/sycl/ext/oneapi/experimental/common_annotated_properties/properties.hpp [hunk 2]
```diff
@@ -0,0 +1,64 @@
+//==-- common_annotated_properties.hpp - SYCL properties associated with
```
Comment (dm-vodopyanov) [inline sycl/include/sycl/ext/oneapi/experimental/common_annotated_properties/properties.hpp:1]:
```suggestion
//==-- properties.hpp - SYCL properties associated with
```

**Comment by dm-vodopyanov** [comment]
@broxigarchen, please update the branch because of many failures in Intel CPU/GEN9 GPU check.

**Review by dm-vodopyanov — APPROVED** [review]

**Comment by dm-vodopyanov** [comment]
@broxigarchen this will be an API break inside one release, or API break between releases?

**Comment by broxigarchen** [comment]
> @broxigarchen this will be an API break inside one release, or API break between releases?

This will be no API break. This is a new feature starting from this release

**Comment by broxigarchen** [comment]
Hi @intel/llvm-gatekeepers can you help to merge this? Thanks!

**Comment by dm-vodopyanov** [comment]
> > @broxigarchen this will be an API break inside one release, or API break between releases?
> 
> This will be no API break. This is a new feature starting from this release

@broxigarchen to double-check, this patch will be cherry-picked to internal repo, correct?

**Comment by broxigarchen** [comment]
> > > @broxigarchen this will be an API break inside one release, or API break between releases?
> > 
> > 
> > This will be no API break. This is a new feature starting from this release
> 
> @broxigarchen to double-check, this patch will be cherry-picked to internal repo, correct?

Yes

File: sycl/include/sycl/ext/intel/experimental/fpga_annotated_properties.hpp [hunk 3]
```diff
@@ -1,4 +1,5 @@
-//==-- properties.hpp - SYCL properties associated with annotated_arg/ptr --==//
+//==-- fpga_annotated_properties.hpp - SYCL properties associated with
+// annotated_arg/ptr --==//
```
Comment (bader) [inline sycl/include/sycl/ext/intel/experimental/fpga_annotated_properties.hpp:2]: @broxigarchen, formatting if off here and in sycl/include/sycl/ext/oneapi/experimental/common_annotated_properties/properties.hpp.

**Pull request merged**